### DISCUSSION TITLE
How can i actually make popper 3kb?

### DISCUSSION THREAD

**Discussion by verybigelephants** [floating-ui/floating-ui#2732] [Q&A] [answered]
Everywhere on popper site and in documentation it says _Weighs just 3 kB!_, but when you actually use the file https://unpkg.com/@popperjs/core@2 it actually weights whooping 20kB, which is really unreasonable for a library that positions elements over other ones

Am i missing something? How can i make it 3kb?

**Comment by atomiks** [comment, marked as answer]
The number refers to the minified+brotli compressed `popper-base` distribution: https://popper.js.org/docs/v2/#distribution-targets. However, note that Popper is no longer actively maintained now — use Floating UI instead (shown on the website's header).

For Floating UI to be as small as possible, use [the core library + create a custom DOM platform](https://floating-ui.com/docs/platform) (**you also need a treeshaking-capable bundler to remove unused code**). The custom platform you create shouldn't need to handle the many edge cases the library handles by default that makes it useful & reliable for production teams.

**Reply by verybigelephants** [reply]
oh well :( i was hoping not to re-invent the wheel looking for a simple umd to display simple tooltips on a very basic website where i am definitely not going into overly complicated frontend build stuff

but now i use 20kb popper+20kb tippy just just to show text when someone hovers over a div. every classic "umd" tooltip javascript library has been updated like 7 years ago or and/or is jquery dependant. and everyone nowdays overcomplicates frontend with builds and vue/react/svelte framework for applications and websites that could've been literally 15kb vanilla js file. what ever happend to frontend development ffs /endrant

if anyone finds this thread and happens to know a very lightweight "tooltip" library in 1-5kb i would be grateful. or by that time i will probably just implement it myself :(

**Reply by atomiks** [reply]
There are CSS tooltip libs that can work for simple cases like balloon.css

**Reply by verybigelephants** [reply]
@atomiks thank you for the suggestion, i have already tried three most popular ones (hint.css/baloon.css/microtip.css) but i really don't think that it's a very seroius solution/may be used only under very non-important tooltip cases; as since it's css only it cannot really correctly tacakle a problem of tooltip overflowing document dom on smaller screens/containers with overflow:hidden and other cases where javascript is necessary for position calculations

**Reply by atomiks** [reply]
Ok well, you seem to realize that "simple" comes with trade-offs, and for it to be robust in production environments to handle various edge cases like the ones you mentioned, it requires more code to work :-) And if you don't use a bundler, you have no option to remove unused code from libraries since they also have to provide a myriad of other features for different use cases for consumers — and the best way for us to support that in a generalized fashion is via module tree-shaking.

I also don't know why you're referencing uncompressed parse sizes instead of sizes transferred over the wire. `@floating-ui/dom` is about 7.5 kB minzipped transferred over the wire with no tree-shaking at all and the tutorial shows you how to create a simple tooltip such that the total is about 8 kB, which is about half the size of Tippy.js.

Anyway, React is not "overcomplicated" if you're working on a large app in a team, it's necessary to produce a good app. If you're just messing around with hobby projects, sure, but that's not how production apps work in the real world

**Reply by verybigelephants** [reply]
`Anyway, React is not "overcomplicated" if you're working on a large app in a team, it's necessary to produce a good app. If you're just messing around with hobby projects, sure, but that's not how production apps work in the real world`

i have been doing fullstack for 17 years now, and have written apps both the oldscool vanilla js/jquery and both in vue and react, so belive me, i know what it means to work on a large app in a team with a reactive framework:) 

i undrstand when you build something really big it's a necessary step (even though i'd still prefer vue over react *looks hatefully at react*), but i still think that vast majority of projects that people do nowdays do not need a reactive framework at all. 
edit: btw, even if you want to go reactive, a great lib for achieving minimal reactive functionality without forcing your users to download huge bundled app js [petite vue](https://github.com/vuejs/petite-vue), awesome spinoff

anyways this thread got really out of topic, thanks for the suggestion again

**Reply by mustafa0x** [reply]
[bootstrap.native/src/util/styleTip.ts](https://github.com/thednp/bootstrap.native/blob/master/src/util/styleTip.ts) is smaller than popper/floating ui and supports the core features (not sure about autoUpdate though). It's not shipped as a stand alone library however. I've extracted it and used it for a few projects.
